### PR TITLE
Use Compound List in more screens.

### PR DESCRIPTION
**Pull request by pixlwave**
Adopts ListRow in the following screens:

| Start Chat | Room Member Details | Notification Settings (minor tweak) | Key Backup (minor tweak) |
| - | - | - | - |
| en-GB-iPhone-14 createRoom | en-GB-iPhone-14 roomMemberDetails | en-GB-iPhone-14 roomNotificationSettingsCustomSetting | test_secureBackupScreen Recovery-incomplete |

### CODE REVIEW

File: UnitTests/__Snapshots__/PreviewTests/test_notificationSettingsEditScreen.Applying-change.png
Comment (pixlwave) [inline UnitTests/__Snapshots__/PreviewTests/test_notificationSettingsEditScreen.Applying-change.png:1]: This one is clearly wrong, came from https://github.com/vector-im/compound-ios/pull/47 where I didn't consider this situation. Need to discuss with @janogarcia the best option to fix it.

File: UITests/Sources/AppLockUITests.swift [hunk 1]
```diff
@@ -87,6 +87,7 @@ class AppLockUITests: XCTestCase {
         
         try await app.assertScreenshot(.appLockFlow, step: Step.unlocked)
         try client.send(.notification(name: UIApplication.didEnterBackgroundNotification))
+        try await Task.sleep(for: .milliseconds(500)) // Don't overwrite the previous signal immediately.
```
Comment (pixlwave) [inline UITests/Sources/AppLockUITests.swift:90]: This is broken on CI.

**Review by nimau — APPROVED** [review]
LGTM (except for the `NotificationSettingsEditScreen` and the `.isWaiting` but I presume it will be fixed in `Compound`)

File: UnitTests/__Snapshots__/PreviewTests/test_notificationSettingsEditScreen.Applying-change.png
Comment (pixlwave) [reply UnitTests/__Snapshots__/PreviewTests/test_notificationSettingsEditScreen.Applying-change.png:1]: Handled in https://github.com/vector-im/compound-ios/pull/48

**Pull request merged**